libexec/python/docker/api.py

```diff
@@ -47,24 +52,46 @@ def create_runscript(cmd,base_dir):
     return output_file
 
 
-def get_token(repo_name,namespace="library",scope="repository",permission="pull"):
+def get_token(registry,repo_name,namespace,scope="repository",permission="pull",auth=None):
     '''get_token will use version 2.0 of Docker's service to return a token with given permission and scope - this
```

Comment:
What is the reason for requiring the registry here? Akin to the other functions, why not have the default as None, and then use api_base if it is? We might want to turn those three lines into a proper function.